sklearn/utils/sparsefuncs_fast.pyx

```diff
@@ -7,18 +7,21 @@
 # License: BSD 3 clause
 
 from libc.math cimport fabs, sqrt, isnan
+from libc.stdint cimport intptr_t
 
 cimport numpy as cnp
 import numpy as np
 from cython cimport floating
+from ..utils._typedefs cimport float64_t, intp_t, uint64_t
 
 cnp.import_array()
 
-ctypedef fused integral:
-    int
-    long long
 
-ctypedef cnp.float64_t DOUBLE
+# TODO: use `{int,float}{32,64}_t` when cython#5230 is resolved:
+# https://github.com/cython/cython/issues/5230
+ctypedef fused integral:
+    int  # int32_t
+    long long  # int64_t
```

Comment:
Would it make sense to move this to `sklearn.utils._typedefs`?

Reply:
`cython.integral` is an already existing predefined fused type for short, int, long. So the name is not good to start with.
I would put it to _typedefs if it can be reused somewhere else => another PR.